rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,727 @@
+              schema:
+                type: boolean
+                description: true if the table was dropped and false if the table did not exist
+    head:
+      tags:
+        - Catalog API
+      summary: Check if a table exists
+      operationId: tableExists
+      description:
+        Check if a table exists within a given namespace. Returns the standard response with `true` when found. Will return a TableNotFound error if not present.
+      parameters:
+        - name: namespace
+          in: path
+          required: true
+          schema:
+            type: string
+        - name: table
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: OK
+        '412':
+          description: Table Not Found
+  /v1/namespaces/{namespace}/tables:
+    parameters:
+      - name: namespace
+        description: Namespace under which to list tables.
+        in: path
+        required: true
+        schema:
+          type: string
+        examples:
+          singlepart_namespace:
+            value: "accounting"
+          multipart_namespace:
+            value: "accounting.tax"
+    get:
+      tags:
+        - Catalog API
+      summary: List all table identifiers underneath a given namespace
+      description: Return all table identifiers under this namespace
+      operationId: listTables
+      parameters:
+        - name: namespace
+          description: Namespace under which to list tables.
+          in: path
+          required: true
+          schema:
+            type: string
+          examples:
+            singlepart_namespace:
+              value: "accounting"
+            multipart_namespace:
+              value: "accounting.tax"
+        # TODO - There's a much more native way to handle pagination
+        - name: limit
+          description: number of values to return in one request
+          in: query
+          required: false
+          schema:
+            type: integer
+          example: 10
+        - name: offset
+          description: Place in the response to continue from if paginating
+          in: query
+          required: false
+          schema:
+            type: integer
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListTablesResponse'
+    post:
+      tags:
+        - Catalog API
+      summary: Create a table with the identifier given in the body
+      operationId: createTable
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateTableRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CreateTableResponse'
+    put:
```

Comment:
Does it make sense to have a namespace in the path for this? Both identifiers in the request are full identifiers. I think it would be more straightforward to have a `POST /rename` route instead of `PUT /namespaces/{namespace}/tables`.

Reply:
I originally had a `POST /tables/renameTable` (similar to what you have here), but I changed it to this format at the request of somebody else.

Happy to change it back. I don't feel that the `namespace` for this path is super clear on the source one and the target one. I'd rather deal with two full identifiers specified in the body.

Reply:
How does `POST /tables/renameTable` sound? I think keeping `tables` in the path is a bit more clear than plain `rename`.

I'll do it that way. Let me know if it should be changed.

Reply:
I think this depends on whether we restructure the API around catalog and table like Ryan's suggestion. Let's come back to this after considering that change.